### ISSUE TITLE
Investigate and remove `try` clause in `xmlToDuration()`

### ISSUE THREAD

**Issue by jacobtylerwalls** [cuthbertLab/music21#967]
I touched this try/except in `xmlToDuration()` merely to unindent it, but I think it can be removed. Test suite doesn't exercise it. And I can't see how there could be a failure to update components on `d` when they have just been created on `durRaw`, since at the present, the setter for `components` clears the duration tuples first. All I can come up with is that maybe a bug is lurking here if tuplets on `d` aren't cleared by the components setter and then they interact with the new components to turn out non-power-of-2 uncomputable things, but that's a bug with regard to not clearing tuplets, not something we should try/except. Worth another PR and some **test cases** at some point. Probably a vestige of an earlier time with some infelicities with the components setter?

_Originally posted by @jacobtylerwalls in https://github.com/cuthbertLab/music21/pull/941#r616228817_

**Comment by jacobtylerwalls** [comment]
Also worth noting is that after #899 this is only reachable in the case of grace notes (the only time we pass a music21 object and therefore `d` is not None).

https://github.com/cuthbertLab/music21/blob/e9165394db519ff4ad6f8b8a0e97cf9fb6daf91f/music21/musicxml/xmlToM21.py#L3259-L3275
